src/tests/ftest/server/cpu_usage.py

```diff
@@ -16,44 +16,48 @@ class CPUUsage(TestWithServers):
     :avocado: recursive
     """
 
-    @skipForTicket("DAOS-5504")
     def test_cpu_usage(self):
         # pylint: disable=pylint-bad-continuation
         """
         JIRA ID: DAOS-4826
         Test Description: Test CPU usage of formatted and idle engine.
         :avocado: tags=all,hw,server,small,full_regression,cpu_usage
         """
-        ps_get_cpu = r"ps -C daos_engine -o %\cpu"
-
-        prev_usage = 1
-        usage = 1
-        time.sleep(5)
-        for _ in range(10):
-            time.sleep(5)
-            task = run_task(hosts=self.hostlist_servers, command=ps_get_cpu)
-            # Sample output.
-            # %CPU
-            # 1798
+        # Get PID of daos_engine with ps.
+        ps_engine = r"ps -C daos_engine -o %\p"
+        pid_found = False
+        # At this point, daos_engine should be started, but do the repetetive
+        # calls just in case.
+        for _ in range(5):
+            task = run_task(hosts=self.hostlist_servers, command=ps_engine)
             for output, _ in task.iter_buffers():
-                usage = str(output).splitlines()[-1]
-                self.log.info("CPU usage = %s", usage)
-            # Check if daos_engine has started.
-            if usage == "%CPU":
-                continue
-
-            usage = int(usage)
-            if usage == 0:
+                self.log.info("ps output = %s", output)
+                pid = str(output).splitlines()[-1]
+                self.log.info("PID = %s", pid)
+                if "PID" not in pid:
+                    pid_found = True
+            if pid_found:
                 break
-            diff = usage - prev_usage
-            diff_p = (float(abs(diff)) / prev_usage) * 100
-
-            # Check if the CPU usage is stable; the change was less than 10%.
-            if diff_p <= float(10):
-                break
-            prev_usage = usage
+            time.sleep(5)
+        if not pid_found:
+            self.fail("daos_engine PID couldn't be obtained!")
+
+        # Wait for CPU usage to stabilize. It usually takes a few sec.
+        time.sleep(10)
+
+        # Get (instantenious) CPU usage of the PID with top.
+        top_pid = "top -p {} -b -n 1".format(pid)
+        usage = -1
+        task = run_task(hosts=self.hostlist_servers, command=top_pid)
+        for output, _ in task.iter_buffers():
+            process_row = str(output).splitlines()[-1]
+            self.log.info("Process row = %s", process_row)
+            values = process_row.split()
+            self.log.info("Values = %s", values)
+            usage = values[8]
```

Comment:
To avoid variations in the `top` output, including but not limited unexpected output that may not yield 9 `values`, maybe we should use something like `ps -o %cpu,pid,cmd -p <pid>`.

Reply:
`ps %cpu` returns the average usage during the lifetime of the process, so it's kind of useless. Since we're using the external tool, I'm not sure how to completely avoid unexpected output. One thing I could do was to add a length check and make sure it has 9 or more columns.